### PR TITLE
Systemtests scale tests - messaging performance

### PR DESCRIPTION
**Pull request by famartinrh**
### Type of change

- Test

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: systemtests/src/main/java/io/enmasse/systemtest/platform/Kubernetes.java [hunk 1]
```diff
@@ -564,13 +568,21 @@ public void deletePod(String namespace, String podName) {
...
+    }
+
+    public void createDeploymentFromResource(String namespace, Deployment resources, long time, TimeUnit unit, boolean doLog) throws Exception {
         if (!deploymentExists(namespace, resources.getMetadata().getName())) {
             Deployment depRes = client.apps().deployments().inNamespace(namespace).create(resources);
             Deployment result = client.apps().deployments().inNamespace(namespace)
-                    .withName(depRes.getMetadata().getName()).waitUntilReady(2, TimeUnit.MINUTES);
-            log.info("Deployment {} created", result.getMetadata().getName());
+                    .withName(depRes.getMetadata().getName()).waitUntilReady(time, unit);
+            if (doLog) {
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/platform/Kubernetes.java]: Is it really necessary to enable/disable logging per method invocation? Does this belong somewhere at the class level or Environment?

File: systemtests/src/main/java/io/enmasse/systemtest/scale/MessagingPerformanceTestManager.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                .toArray(String[]::new);
+        var clientConfig = clientProvider.get();
+        clientConfig.setClientType(ScaleTestClientType.messaging);
+        clientConfig.setAddressesType(type);
+        clientConfig.setAddresses(addressesOfType);
+        clientConfig.setLinksPerConnection(linksPerConnection);
+
+        SystemtestsKubernetesApps.deployScaleTestClient(kubernetes, clientConfig);
+
+        int connectionsInThisClient = (addressesOfType.length/linksPerConnection) * 2; // *2 because client creates sender and receiver
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/scale/MessagingPerformanceTestManager.java]: We should make this configurable, i.e. for the IoT case where you might have a lot more senders than receivers.

File: systemtests/src/main/java/io/enmasse/systemtest/scale/ScaleTestClient.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.scale;
+
+import io.enmasse.systemtest.model.address.AddressType;
+import io.enmasse.systemtest.scale.metrics.ScaleTestClientMetricsClient;
+
+public class ScaleTestClient <X extends ScaleTestClientMetricsClient>{
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/scale/ScaleTestClient.java]: I think the convention is to use T as the first generic parameter, we should stick to that.

File: systemtests/src/main/java/io/enmasse/systemtest/scale/metrics/MetricsAssertions.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.scale.metrics;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Assertions;
+
+public class MetricsAssertions<X> implements Predicate<X> {
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/scale/metrics/MetricsAssertions.java]: X -> T

File: systemtests/src/main/java/io/enmasse/systemtest/time/TimeMeasuringSystem.java [hunk 5]
```diff
@@ -37,6 +37,8 @@
     private String testClass;
     private String testName;
 
+    private boolean logResults = true;
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/time/TimeMeasuringSystem.java]: Is it better enabled/disabled via Environment.java ?

File: systemtests/src/main/java/io/enmasse/systemtest/platform/Kubernetes.java (hunk 1 above)
Comment (famartinrh) [reply systemtests/src/main/java/io/enmasse/systemtest/platform/Kubernetes.java]: with this changes I only disabled the logging in a few places where it was too verbose for the scale test, I'm not sure I want to disable logging for this full class, altough I agree this is not the most elegant solution

File: systemtests/src/main/java/io/enmasse/systemtest/scale/MessagingPerformanceTestManager.java (hunk 2 above)
Comment (famartinrh) [reply systemtests/src/main/java/io/enmasse/systemtest/scale/MessagingPerformanceTestManager.java]: that will require to implement some login in the client, but it's ok and I like it, so I'll add it to my TODO list

File: systemtests/src/main/java/io/enmasse/systemtest/time/TimeMeasuringSystem.java (hunk 5 above)
Comment (famartinrh) [reply systemtests/src/main/java/io/enmasse/systemtest/time/TimeMeasuringSystem.java]: agree, but first need support to set that env var in our pipelines

**Comment by famartinrh** [comment]
@lulf yes I would prefer that

**Pull request merged**